sdktests/client_side_auto_env_attributes.go

```diff
@@ -0,0 +1,299 @@
+package sdktests
+
+import (
+	"strings"
+	"time"
+
+	"github.com/launchdarkly/go-sdk-common/v3/ldcontext"
+	m "github.com/launchdarkly/go-test-helpers/v2/matchers"
+
+	"github.com/launchdarkly/sdk-test-harness/v2/data"
+	"github.com/launchdarkly/sdk-test-harness/v2/framework/harness"
+	"github.com/launchdarkly/sdk-test-harness/v2/framework/ldtest"
+	"github.com/launchdarkly/sdk-test-harness/v2/framework/opt"
+	"github.com/launchdarkly/sdk-test-harness/v2/servicedef"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func doClientSideAutoEnvAttributesTests(t *ldtest.T) {
+	t.RequireCapability(servicedef.CapabilityAutoEnvAttributes)
+	t.Run("events", doClientSideAutoEnvAttributesEventsTests)
+	t.Run("headers", doClientSideAutoEnvAttributesHeaderTests)
+
+	// server-side SDKs do not send user properties in polling/streaming requests
+	if t.Capabilities().Has(servicedef.CapabilityClientSide) {
```

Comment:
Is this needed since these are already under a `doClientSide...` named test? Seems like we are probably guarding against this upstream arleady.